### PR TITLE
chombo-xsdk package

### PR DESCRIPTION
**Pull request by drhansj**
We've completed the chombo-xsdk package (different from the chombo package that Mark Miller created) and tested it for compatibility with xsdk-0.2.0. Building against xsdk PETSC works on Mac, Linux ... still testing on Edison.

### CODE REVIEW

**Comment by tgamblin** [comment]
@drhansj @tjligocki: The Travis build seems to be erroring today for some reason to do with the doc packaging, so ignore the unit test failures.  But there are also `flake8` failures that need to be addressed:

* https://travis-ci.org/LLNL/spack/jobs/232921500

If you could address those while I work out the rest of the test issues that would be great.  See the [guide to contributing](https://spack.readthedocs.io/en/latest/contribution_guide.html) for details.

Also, should this package not just be called `chombo` and merged with @markcmiller86's work?

**Comment by BarrySmith** [comment]
Built successfully on my Mac.

**Comment by adamjstewart** [comment]
@tgamblin I've already started hacking on Travis. See #4266 and #4243.

**Comment by drhansj** [comment]
I'll work on the formatting stuff and re-issue the pull request.

**Comment by tgamblin** [comment]
@drhansj: why close this?

**Comment by drhansj** [comment]
Superceded by https://github.com/LLNL/spack/pull/4268

**Comment by tgamblin** [comment]
@drhansj: why not just push it here, to the same PR?

**Comment by tgamblin** [comment]
It actually looks like the commits you added already showed up here, as you pushed them to your fork (see 70daa4a) above.  You don't have to open a new PR for each change.

**Comment by adamjstewart** [comment]
Worse yet, opening a new PR for each change means that any time someone comments on your code, those comments don't follow the PR and are lost forever.

**Comment by drhansj** [comment]
Sorry, I wanted to make sure the scope was different (changed xsdk package
this time).

**Comment by tgamblin** [comment]
@drhansj: do you mind closing the other PR and reopening here?  I could also do it.  It's fine to modify two packages with related changes in the same PR

**Comment by drhansj** [comment]
Done!

**Comment by drhansj** [comment]
Reopened per @tgamblin request - please make sure to merge in 70daa4

**Comment by BarrySmith** [comment]
building all of xsdk@xsdk-0.2.0 worked successfully on my Apple laptop. Including building Chombo. @jwillenbring

**Comment by drhansj** [comment]
Works on edison, too, if you remove the steps that run tests, examples on installation. New commit fbf90cd

File: var/spack/repos/builtin/packages/chombo-xsdk/package.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        if spec.variants['mpi']:
+            options.append('RUN=%s -np 2 ./' %
+                           join_path(spec['mpi'].prefix.bin, 'mpirun'))
+        else:
+            options.append('RUN=./')
+
+        if spec.variants['use_hdf']:
+            options.append('HDFINCFLAGS=-I%s/include' % spec['hdf5'].prefix)
+            options.append('HDFLIBFLAGS=-L%s/lib -lhdf5 -lz' %
+                           spec['hdf5'].prefix)
```
Comment (davydden) [inline var/spack/repos/builtin/packages/chombo-xsdk/package.py]: try `spec['hdf5'].libs.ld_flags()`  to get `-L../lib -lhdf5`.

File: var/spack/repos/builtin/packages/chombo-xsdk/package.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            options.append('HDFMPIINCFLAGS=-I%s/include' % spec['hdf5'].prefix)
+            options.append('HDFMPILIBFLAGS=-L%s/lib -lhdf5 -lz' %
```
Comment (davydden) [inline var/spack/repos/builtin/packages/chombo-xsdk/package.py]: same as above

File: var/spack/repos/builtin/packages/chombo-xsdk/package.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+                           spec['hdf5'].prefix)
+
+        options.append('syslibflags=%s %s' %
+                       (spec['lapack'].libs, spec['blas'].libs))
```
Comment (davydden) [inline var/spack/repos/builtin/packages/chombo-xsdk/package.py]: you should add those to remove duplicates, i.e. 
```
math_libs = spec['lapack'].libs + spec['blas'].libs
options.append('syslibflags=%s' % math_libs)
```

**Comment by markcmiller86** [comment]
> @davydden ...is this still needed or it's the same as Chombo?

Yes, pretty sure. Chombo and Chombo-xsdk are not compile or link-time compatible. Chombo-xsdk includes numerous tiny changes to class names, etc. (I think that were deemed essential for xsdk compliance though drivers escape me now). @tjligocki or @drhansj would know better.

**Comment by davydden** [comment]
> Chombo-xsdk includes numerous tiny changes to class names, etc

if the build system is exactly the same, i think it would make sense to derive `Chombo-xsdk` from `Chombo` to avoid all code duplication in setup.

File: var/spack/repos/builtin/packages/chombo-xsdk/package.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    # Find out more about Chombo
+    homepage = "http://chombo.lbl.gov/"
+    url = "http://bitbucket.org/drhansj/chombo-xsdk/get/xsdk-0.2.0a.tar.bz2"
+
+    # Versions available
+    version('xsdk-0.2.0', git='http://bitbucket.org/drhansj/chombo-xsdk.git', tag='xsdk-0.2.0')
+    version('develop', git='http://bitbucket.org/drhansj/chombo-xsdk.git', tag='master')
+
+    # Build options/variants
+    variant('dim', default=2, description='Set the physical dimension', values=isIntegral)
```
[the remark this answers is not in the document]
Comment (markcmiller86) [reply var/spack/repos/builtin/packages/chombo-xsdk/package.py]: In Chombo, valid dims are 1,2,3,4,5 and 6. So, this should be a multi-valued variant as in Chombo package. Also, my impression is that default of 3 is "better" in the sense that I think 3D is more common than 2D.

**Comment by markcmiller86** [comment]
> @davydden ...if the build system is exactly the same, i think it would make sense to derive Chombo-xsdk from Chombo to avoid all code duplication in setup.

Well, the build interfaces too are different. Chombo-xsdk build interface is defined to satisfy xsdk compliance in build interface options.

**Comment by alalazo** [comment]
Is this still of interest?

**Comment by markcmiller86** [comment]
Sheesh, wasn't aware I had self-added myself for review here. Is my review necessary here? I don't think it is. I think as long as someone on xsdk approves it, that should be fine. I guess a related question...is the old spack chombo package still needed or has it already been removed.

**Comment by tgamblin** [comment]
@markcmiller86: we took your `chombo` package instead of this one, and Chombo didn't end up in `xsdk`. So unless you feel like reconciling this with your package (or unless someone else does), I'm inclined to close it.  Please feel free to reopen if this still needs to be considered.

**Pull request closed without merging**